primitives/core/src/lib.rs

```diff
@@ -194,6 +195,37 @@ impl<B: BlockT> ParachainBlockData<B> {
 	}
 }
 
+/// Type of recovery to trigger.
+#[derive(Debug, PartialEq)]
+pub enum RecoveryKind {
+	/// Single block recovery.
+	Simple,
+	/// Full ancestry recovery.
+	Full,
+}
+
+/// Structure used to trigger an explicit recovery request via `PoVRecovery`.
+pub struct RecoveryRequest<Block: BlockT> {
+	/// Hash of the last block to recover.
+	pub hash: Block::Hash,
+	/// Recovery delay range. Randomizing the start of the recovery within this interval
+	/// can be used to prevent self-DOSing if the recovery request is part of a
+	/// distributed protocol and there is the possibility that multiple actors are
+	/// requiring to perform the recovery action at approximately the same time.
+	pub delay: RecoveryDelay,
+	/// Recovery type.
+	pub kind: RecoveryKind,
+}
+
+/// The delay between observing an unknown block and triggering the recovery of a block.
+#[derive(Clone, Copy)]
+pub struct RecoveryDelay {
+	/// Start recovering after `min` delay.
+	pub min: Duration,
+	/// Start recovering before `max` delay.
+	pub max: Duration,
+}
+
```

Comment:
Do we really need to have them defined here? I think they should stay in `pov-recovery` and then we import this crate in `consensus/common`

Reply:
At the moment not really, as you correctly suggested we can just import `pov-recovery`in `consensus/common`.

The rationale behind moving the basic types of block recovery out of PoCRecovery was to abstract the recovery mechanism.

When the consensus code finds that there is a new **best** it will trigger a recovery by sending a message (`RecoveryRequest`) via an mpsc channel. The recovery action can then **potentially** happen via any mean  (dummy examples: torrent, ipfs, a distributed filesystem, local filesystem, ...).

 I know, this is not the case **now**. But in this way we don't bound cumulus user of our consensus  to one possibility.

Reply:
Ok. I will put these structs it in the `pov-recovery` crate in the other PR